plutus-core/plutus-ir/src/PlutusIR/Transform/Rename.hs

```diff
@@ -198,30 +198,28 @@ recursive and non-recursive data types with a single function.
 
 instance PLC.HasUniques (Term tyname name uni fun ann) => PLC.Rename (Term tyname name uni fun ann) where
     -- See Note [Marking]
-    rename = through markNonFreshTerm >=> PLC.runRenameT . renameTermM
+    rename = through markNonFreshTerm >=> PLC.liftQuote . PLC.runRenameM . renameTermM
 
 instance PLC.HasUniques (Term tyname name uni fun ann) => PLC.Rename (Program tyname name uni fun ann) where
     rename (Program ann term) = Program ann <$> PLC.rename term
 
 -- See Note [Renaming of constructors].
 -- | A wrapper around a function restoring some old context of the renamer.
-newtype Restorer m = Restorer
-    { unRestorer :: forall a. m a -> m a
+data Restorer = Restorer
+    { unRestorer :: forall a. PLC.ScopedRenameM a -> PLC.ScopedRenameM a
     }
```

Comment:
Not sure about this one. Can be improved somehow?

Reply:
I think it's okay? Why the change to data?